<launch>

    <!-- defines name of the namespace of the drone -->
    <arg name="UAV_NAME" default="$(env UAV_NAME)"/>
    <arg name="RUN_TYPE" default="$(env RUN_TYPE)"/>

    <!-- set to false if the nodelet should run under an external nodelet manager -->
    <arg name="standalone" default="true"/>

    <!-- allows disabling the gui from command line -->
    <arg name="gui" default="true"/>

    <!-- will it run using GNU debugger? -->
    <arg name="debug" default="false"/>
    <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
    <arg if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

    <!-- will it run as standalone nodelet or using a nodelet manager? -->
    <arg name="manager" default="$(arg UAV_NAME)_uav_camera_localization_manager"/>
    <arg if="$(arg standalone)" name="nodelet" value="standalone"/>
    <arg unless="$(arg standalone)" name="nodelet" value="load"/>
    <arg if="$(arg standalone)" name="nodelet_manager" value=""/>
    <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>

    <arg name="stereopair_base_frame_id" default="$(arg UAV_NAME)/basler_stereopair/base"/>
    <arg name="camera1_fleft_pose" default="/$(arg UAV_NAME)/basler_stereopair/camera1_fleft_pose"/>
    <arg name="camera2_fright_pose" default="/$(arg UAV_NAME)/basler_stereopair/camera2_fright_pose"/>
    <!-- Namespace - all topics and parameters will be remapped using this namespace as a prefix (eg.: "/vision_example/image_in" to "/uav1/vision_example/image_in") -->
    <group ns="$(arg UAV_NAME)">

        <!-- ClASS_NAME nodelet -->
        <node pkg="nodelet"
              type="nodelet"
              name="camera_localization"
              args="$(arg nodelet) camera_localization/CameraLocalization $(arg nodelet_manager)"
              launch-prefix="$(arg launch_prefix_debug)"
              output="screen">

            <!-- Parameters loaded from launch file -->
            <param name="gui" type="bool" value="$(arg gui)"/>

            <param name="UAV_NAME" type="string" value="$(arg UAV_NAME)"/>
            <param name="base_frame_pose" type="string" value="$(arg stereopair_base_frame_id)"/>
            <param name="m_name_CL" type="string" value="uav1/basler_left_optical"/>
            <param name="m_name_CR" type="string" value="uav1/basler_right_optical"/>
            <param name="camera1_fleft_pose" type="string" value="$(arg camera1_fleft_pose)"/>
            <param name="camera2_fright_pose" type="string" value="$(arg camera2_fright_pose)"/>
            <rosparam file="$(find uav_camera_localization)/config/general.yaml"/>
            <rosparam file="$(find uav_camera_localization)/config/cameras_roi.yaml"/>

            <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->

            <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->

        </node>

    </group>

</launch>
